### ISSUE TITLE
Main toolbar shows stop button when app run with debugger has stopped

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#7560]
Do not have a specific repro. Was debugging Main.sln and then at some point the run button remains as a stop button. This has happened twice now on different VS Mac 8.1 builds. The IDE log has an error:

```
An unhandled exception has occurred. Terminating Visual Studio? False
System.ArgumentNullException: Value cannot be null.
Parameter name: toolTip
  at AppKit.NSSegmentedCell.SetToolTip (System.String toolTip, System.nint forSegment) [0x0005e] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.6.0.25/src/Xamarin.Mac/AppKit/NSSegmentedCell.g.cs:502 
  at MonoDevelop.MacIntegration.MainToolbar.ButtonBar.UpdateButton (MonoDevelop.Components.MainToolbar.IButtonBarButton button, System.Int32 idx) [0x0004b] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MacPlatform/MainToolbar/ButtonBar.cs:217 
  at MonoDevelop.MacIntegration.MainToolbar.ButtonBar.RebuildSegments () [0x00068] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MacPlatform/MainToolbar/ButtonBar.cs:203 
  at MonoDevelop.MacIntegration.MainToolbar.ButtonBar.<.ctor>b__5_3 (System.Object o, System.EventArgs e) [0x00000] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MacPlatform/MainToolbar/ButtonBar.cs:133 
  at MonoDevelop.Components.MainToolbar.MainToolbarController+ButtonBarButton.Update (MonoDevelop.Components.Commands.CommandInfo ci) [0x000b9] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/MainToolbarController.cs:946 
  at MonoDevelop.Components.MainToolbar.MainToolbarController+ButtonBarButton.Update () [0x00078] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/MainToolbarController.cs:922 
  at MonoDevelop.Components.MainToolbar.MainToolbarController.MonoDevelop.Components.Commands.ICommandBar.Update (System.Object activeTarget) [0x00081] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/MainToolbarController.cs:866 
  at MonoDevelop.Components.Commands.CommandManager.UpdateToolbars () [0x0002c] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Commands/CommandManager.cs:2505 
  at MonoDevelop.Components.Commands.CommandManager.UpdateStatus () [0x00010] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Commands/CommandManager.cs:2357 
  at GLib.Timeout+TimeoutProxy.HandlerInternal (System.IntPtr data) [0x00017] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/gtk-sharp-None/glib/Timeout.cs:38 


Error while executing command: Start Debugging
System.ArgumentNullException: Value cannot be null.
Parameter name: toolTip
  at AppKit.NSSegmentedCell.SetToolTip (System.String toolTip, System.nint forSegment) [0x0005e] in /Library/Frameworks/Xamarin.Mac.framework/Versions/5.6.0.25/src/Xamarin.Mac/AppKit/NSSegmentedCell.g.cs:502 
  at MonoDevelop.MacIntegration.MainToolbar.ButtonBar.UpdateButton (MonoDevelop.Components.MainToolbar.IButtonBarButton button, System.Int32 idx) [0x0004b] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MacPlatform/MainToolbar/ButtonBar.cs:217 
  at MonoDevelop.MacIntegration.MainToolbar.ButtonBar.RebuildSegments () [0x00068] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MacPlatform/MainToolbar/ButtonBar.cs:203 
  at MonoDevelop.MacIntegration.MainToolbar.ButtonBar.<.ctor>b__5_3 (System.Object o, System.EventArgs e) [0x00000] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/addins/MacPlatform/MainToolbar/ButtonBar.cs:133 
  at MonoDevelop.Components.MainToolbar.MainToolbarController+ButtonBarButton.Update (MonoDevelop.Components.Commands.CommandInfo ci) [0x000b9] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/MainToolbarController.cs:946 
  at MonoDevelop.Components.MainToolbar.MainToolbarController+ButtonBarButton.Update () [0x00078] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/MainToolbarController.cs:922 
  at MonoDevelop.Components.MainToolbar.MainToolbarController.MonoDevelop.Components.Commands.ICommandBar.Update (System.Object activeTarget) [0x00081] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/MainToolbarController.cs:866 
  at MonoDevelop.Components.Commands.CommandManager.UpdateToolbars () [0x0002c] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Commands/CommandManager.cs:2505 
  at MonoDevelop.Components.Commands.CommandManager.DispatchCommand (System.Object commandId, System.Object dataItem, System.Object initialTarget, MonoDevelop.Components.Commands.CommandSource source, System.Nullable`1[T] time, MonoDevelop.Components.Commands.CommandInfo sourceUpdateInfo) [0x003ea] in /Users/vsts/agent/2.150.3/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Commands/CommandManager.cs:1601 
```

Build Information
Release ID: 801002462
Git revision: 29162fa2ff21bf2018bd5d3a7f60866fb62d547b
Build date: 2019-05-10 21:38:59+00
Build branch: release-8.1
Xamarin extensions: 735d3d5d1daaee9cf9b045febed581680960e8b2

> VS bug [#892065](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/892065)

**Comment by Therzok** [comment]
Filed a bug against xamarin-macios: https://github.com/xamarin/xamarin-macios/issues/6044

We can either send the selector directly or pass string.Empty.

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-13] Comment by **xamarinc**



Fixed in version 8.2.0.385 (master)



Pull Request #7622 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 535db983fc1cba8e449c414a246f307f18da4c53 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-14] Comment by **xamarinc**



Fixed in version 8.1.0.2532 (release-8.1)



Pull Request #7623 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 90085d9beb9e98cd15ba9caa256d4095a892dc50 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-23] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #7622 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 535db983fc1cba8e449c414a246f307f18da4c53 (xamarin/monodevelop)
